### PR TITLE
Enable SDL validation in rel/3.1

### PR DESCRIPTION
**Pull request by wtgodbe**
Part of aspnet/AspNetCore-Internal#3293

@riarenas @JohnTortugo PTAL

CC @aspnet/build

### CODE REVIEW

**Comment by wtgodbe** [comment]
@sunandabalu PTAL

File: azure-pipelines.yml [hunk 1]
```diff
@@ -13,6 +13,10 @@ variables:
     value: https://localhost:8081
   - name: _CosmosToken
     value: C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==
+    
+  # used for post-build phases, internal builds only
+  - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+    - group: DotNet-Winforms-SDLValidation-Params
```
Comment (sunandabalu) [inline azure-pipelines.yml]: AspNet should create its own variable group as the SDL settings are subjective to the repo. This variable group has WinForms specific values and would not work for AspNet.

Comment (wtgodbe) [reply azure-pipelines.yml]: Where can I create this variable group/look at the winforms one for an example?

Comment (riarenas) [reply azure-pipelines.yml]: https://dev.azure.com/dnceng/internal/_library?itemType=VariableGroups&view=VariableGroupView&variableGroupId=66&path=DotNet-Winforms-SDLValidation-Params for an example

**Comment by wtgodbe** [comment]
This is equivalent to aspnet/Extensions#2579 which was validated & approved

**Pull request merged**